src/libraries/System.Formats.Tar/tests/TarFile/TarFile.ExtractToDirectoryAsync.Stream.Tests.cs

```diff
@@ -97,6 +97,19 @@ await using (TarWriter writer = new TarWriter(archive, TarEntryFormat.Ustar, lea
             }
         }
 
+        [Fact]
+        public async Task ExtractEntry_DockerImageTarWithFileTypeInDirectoriesInMode_SuccessfullyExtracts_Async()
```

Comment:
I would also include a version of this test using a compressed archive since that's what the customer reported. Also, add the sync test. But this is up to you, not a big deal tbh.

Reply:
That will require setting up token to fetch gzipped tar archive from docker api and posting a PR like https://github.com/dotnet/runtime-assets/pull/279. I don't have access to those tokens (and don't think it is super important for 7.0 minimal fix).